### PR TITLE
Merchant new item #14

### LINKED ISSUE #13: User Story 56, Merchant cannot add an item if details are bad/missing
```
As a merchant
When I try to add a new item
If any of my data is incorrect or missing (except image)
Then I am returned to the form
I see one or more flash messages indicating each error I caused
All fields are re-populated with my previous data
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #19

child of #79

depends on #14

### LINKED ISSUE #14: User Story 55, Merchant adds an item
```
As a merchant
When I visit my items page
And I click on the link to add a new item
My URI route should be "/dashboard/items/new"
I see a form where I can add new information about an item, including:
- the name of the item, which cannot be blank
- a description for the item, which cannot be blank
- a thumbnail image URL string, which CAN be left blank
- a price which must be greater than $0.00
- my current inventory count of this item which is 0 or greater

When I submit valid information and save the form
I am taken back to my items page
I see a flash message indicating my new item is saved
I see the new item on the page, and it is enabled and available for sale
If I left the image field blank, I see a placeholder image for the thumbnail
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #19

child of #79

depends on #18

### PR DESCRIPTION
**Pull request by allieIsOkay**
- Adds tests for merchants ability to add new items
- Adds testing for ability to leave item image field blank when generating new items
- Adds testing for display of errors for each required form field
- Adds ability for merchant to add items
- Adds default image override when input is empty
- Updates routing to match spec
- closes #14 
- closes #13

### CODE REVIEW

**Review by timnallen — APPROVED** [review]
Love the refactoring. Flash message is great and error messages on the page. Let's do it this way for all our forms.

File: app/controllers/merchant/items_controller.rb [hunk 1]
```diff
@@ -2,4 +2,26 @@ class Merchant::ItemsController < Merchant::BaseController
...
+    else
+      flash[:danger] = "There are problems with the provided information."
+      render :new
+    end
+  end
+
+  private
+
+  def item_params
+    params[:item][:image] = "https://via.placeholder.com/200x300?text=LittleShop" if params[:item][:image] == ""
```
Comment (csvlewis) [inline app/controllers/merchant/items_controller.rb:43]: Nice workaround for the placeholder image!

**Review by csvlewis — APPROVED** [review]
This looks great! I agree with Tim, we should try to implement the dynamic error message function wherever we can in our forms.

**Review by teresa-m-knowles — APPROVED** [review]
This looks great, just resolved the merge conflicts
